c_src/eleveldb.cc

```diff
@@ -1280,7 +1336,7 @@ streaming_start(ErlNifEnv * env,
         work_item = new RangeScanTask(env, reply_ref, db_ptr.get(),
                                  start_key, end_key_ptr, opts, sync_handle->sync_obj_);
     } catch(std::runtime_error& err) {
```

Comment:
Nightmare.  Hope this is coded correctly or Erlang crashes.  There is a reason for no try/catch blocks.